C7GameData/ImportCiv3.cs

```diff
@@ -68,11 +61,65 @@ static public C7SaveFormat ImportSav(string savePath, string defaultBicPath)
             return c7Save;
         }
 
-        // stub
-        static public C7RulesFormat ImportBic()
-        {
-            C7RulesFormat c7Save = new C7RulesFormat();
-            return c7Save;
-        }
-    }
+		/**
+		 * defaultBiqPath is used in case some sections (map, rules, player data) are not
+		 * present.
+		 */
+		public static C7SaveFormat ImportBiq(string biqPath, string defaultBiqPath)
+		{
+			C7SaveFormat c7Save = new C7SaveFormat();
+			
+			byte[] biqBytes = Util.ReadFile(biqPath);
+			BiqData theBiq = new BiqData(biqBytes);
+			
+			ImportCiv3TerrainTypes(theBiq, c7Save);
+			SetMapDimensions(c7Save, theBiq);
+			
+			//Import tiles.  This is different from the SAV version as we have only BIQ TILE objects.
+			int i = 0;
+			foreach (QueryCiv3.Biq.TILE civ3Tile in theBiq.Tile)
+			{
+				Civ3ExtraInfo extra = new Civ3ExtraInfo
+				{
+					BaseTerrainFileID = civ3Tile.TextureFile,
+					BaseTerrainImageID = civ3Tile.TextureLocation,
+				};
+				int y = i / (theBiq.Wmap[0].Width / 2);
+				int x = (i % (theBiq.Wmap[0].Width / 2)) * 2 + (y % 2);
```

Comment:
is one of these supposed to use Height?

Reply:
Nope, that's correct.  The x uses modulus instead of division, so e.g. if i = 73 and width = 100, it will be 73 % 50 = 23.  Then times 2 = 46.  Plus y %2 = 1, X = 47, Y = 1 would be the 73rd tile (base zero) in the list.

Definitely not intuitive with the isometric grid, but it appears to be correct (other than using BIQ data for the width, it's actually copied from code Lucien wrote in December).

It probably does make sense to have it be shared across the two areas though; I'm adding a shared function now.